services/QuillLMS/spec/factories/auth_credentials.rb

```diff
@@ -30,7 +30,20 @@
     expires_at 1.day.from_now
     user
 
-    factory :canvas_auth_credential, parent: :auth_credential, class: :CanvasAuthCredential
+    factory :canvas_auth_credential, parent: :auth_credential, class: :CanvasAuthCredential do
```

Comment:
This factory's complexity is related to the associated `canvas_instance_auth_credential` record, which in production should be created after a `canvas_auth_credential` creation.  However, this creates a circularity when unit testing `canvas_instance_auth_credential` hence the `canvas_auth_credential_without_canvas_instance_auth_credential` factory.  I'm not sure of a cleaner way to do this as factory bot doesn't seem to allow for conditional callbacks and I don't want to introduce two new traits for default behavior.  I'm definitely open to suggestions here.

Reply:
This is an awkward compromise, but I can't think of a cleaner way to do it offhand.